src/components/NamePicker.tsx

```diff
@@ -1,10 +1,76 @@
-import React from 'react';
-import TextField from '@material-ui/core/TextField';
+import React, { useState, MouseEvent, ChangeEvent } from 'react';
 import Grid from '@material-ui/core/Grid';
+import Popover from '@material-ui/core/Popover';
+import TextField from '@material-ui/core/TextField';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
 import { makeStyles } from '@material-ui/core/styles';
 
 export default function NamePicker(): JSX.Element {
   const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [anchorEl2, setAnchorEl2] = useState<null | HTMLElement>(null);
+  const [disabled1, setDisabled1] = useState(true);
+  const [disabled2, setDisabled2] = useState(true);
+  const [firstName, setFirstName] = useState({
+    firstSounds: false,
+    firstSimilar: false,
+    firstInitials: false,
+  });
+
+  const [lastName, setLastName] = useState({
+    lastSounds: false,
+    lastSimilar: false,
+    lastInitials: false,
+  });
+
+  const handleClick1 = (event: MouseEvent<HTMLElement>): void => {
+    setAnchorEl(event.currentTarget);
+    if (anchorEl === null) {
```

Comment:
This can be written simply as `if (!anchorEl) {`. Since TS knows `anchorEl` has to be either an `HTMLElement` or `undefined`, the `!` check is unambiguous and will always do the right thing (since `HTMLElement` is an object).

Reply:
Updated and thanks!